components/services/ipfs/ipfs_service_impl.cc

```diff
@@ -108,22 +109,62 @@ bool LaunchProcessAndExit(const base::FilePath& path,
   return true;
 }
 
+void MonitorChild(base::ProcessHandle p_handle,
+                  base::OnceCallback<void(base::ProcessId)> callback) {
+#if defined(OS_POSIX)
+  char buf[PIPE_BUF];
+
+  while (1) {
+    if (read(pipehack[0], buf, sizeof(buf)) > 0) {
+      pid_t pid;
+      int status;
+
+      if ((pid = waitpid(-1, &status, WNOHANG)) != -1) {
```

Comment:
instead of waiting on all children (`-1`), pass the p_handle of the child you are monitoring?

Reply:
fixed in https://github.com/brave/brave-core/pull/8620/commits/7757acb9179b6e2c366625d8a61168bd0f106746